packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart

```diff
@@ -0,0 +1,713 @@
+  /// save and restore its scroll position in the [PageStorage] (see
+  /// [ScrollController.keepScrollOffset]). It can be used to read the current
+  /// scroll position (see [ScrollController.offset]), or change it (see
+  /// [ScrollController.animateTo]).
+  final ScrollController controller;
+
+  /// How the scroll view should respond to user input.
+  ///
+  /// For example, determines how the scroll view continues to animate after the
+  /// user stops dragging the scroll view.
+  ///
+  /// Defaults to matching platform conventions.
+  final ScrollPhysics physics;
+
+  /// {@macro flutter.widgets.wheelList.diameterRatio}
+  ///
+  /// Defaults to an arbitrary but aesthetically sane number of 2.0.
+  ///
+  /// Must not be null and must be positive.
+  final double diameterRatio;
+
+  /// Perspective of the cylindrical projection.
+  ///
+  /// A number between 0.01 and 0 where 0 means looking at the cylinder from
+  /// infinitely far with an infinitely small field of view and 1 means looking
+  /// at the cylinder from infinitely close with an infinitely large field of
+  /// view (and also un-renderable).
+  ///
+  /// Defaults to an arbitrary but aesthetically sane number of 0.003. A larger
+  /// number brings the vanishing point closer and a smaller number pushes the
+  /// vanishing point further.
+  ///
+  /// Must not be null.
+  final double perspective;
+
+  /// Size of all children in the main axis. Must not be null and must be
+  /// positive.
+  final double itemExtent;
+
+  /// Whether to clip painted children to the inside of this scroll view.
+  ///
+  /// Defaults to [true]. Must not be null.
+  ///
+  /// If this is false and [renderChildrenOutsideViewport] is true, the
+  /// first and last children may extend outside.
+  final bool clipToSize;
+
+  /// Whether to paint children inside the viewport only.
+  ///
+  /// If false, every child will be painted. However the [Scrollable] is still
+  /// the size of the viewport and detects gestures inside only.
+  ///
+  /// Defaults to [false]. Must not be null. Cannot be true if [clipToSize]
+  /// is also true since children outside the viewport will be clipped, leading
+  /// to gratuitous waste.
+  final bool renderChildrenOutsideViewport;
+
+  /// List of children to scroll on top of the cylinder.
+  final List<Widget> children;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scrollable(
+      controller: controller,
+      physics: physics,
+      viewportBuilder: (BuildContext context, ViewportOffset offset) {
+        return new _ListWheelViewport(
+          diameterRatio: diameterRatio,
+          perspective: perspective,
+          itemExtent: itemExtent,
+          clipToSize: clipToSize,
+          renderChildrenOutsideViewport: renderChildrenOutsideViewport,
+          offset: offset,
+          children: children,
+        );
+      },
+    );
+  }
+}
+
+class _ListWheelViewport extends MultiChildRenderObjectWidget {
```

Comment:
this should probably be public, in case someone wants to use it in different contexts

Reply:
I think it's an easy change for anyone who wants to use it publicly (I kinda want to avoid copy pasting all the fields' docs again)

Reply:
We usually have a 1:1 correspondence between render object and render object widget. I think it would be weird for this to be an exception.

Reply:
Actually we have two patterns here.

The first is to have the render object in the rendering/ layer and the render object widget in the widgets/ layer, both public.

The second is to have them both in the material/ (or cupertino/) layer, both private, in the same file that implements the widget in question. (We have several examples of highly-specialised render objects in those layers.)

If you think this class is not something that will ever be reused outside of the context of the other widget, then you could just make it private. But if you think it's valuable to allow people to use the render object directly (e.g. because they might want to make a new widget with similar effects, or because they might want this effect even if they don't use the widgets layer), then you should put the render object in the rendering layer, and the render object widget in `basic.dart`.

Reply:
I moved some stuff around. Mostly following the pattern Viewport/RenderViewport used. 

Moved the RenderListWheelViewport to rendering.